localstack/services/lambda_/provider.py

```diff
@@ -1665,21 +1665,36 @@ def create_event_source_mapping(
             )
 
         state = lambda_stores[context.account_id][context.region]
-        function_name = request["FunctionName"]
-
-        if api_utils.FULL_FN_ARN_PATTERN.match(function_name):
-            fn_arn = function_name
-            function_name = api_utils.get_function_name(function_name, context)
-        else:
-            fn_arn = api_utils.unqualified_lambda_arn(
-                function_name, context.account_id, context.region
-            )
+        request_function_name = request["FunctionName"]
+        # can be either a partial arn or a full arn for the version/alias
+        function_name, qualifier, account, region = function_locators_from_arn(
+            request_function_name
+        )
+        account = account or context.account_id
```

Comment:
FYI: we have not tested cross-account behavior for create operations in the Lambda core API yet